### PR TITLE
Add type annotations to namespace.py

### PR DESCRIPTION
**Pull request by xinrong-databricks**

### CODE REVIEW

File: databricks/koalas/namespace.py [hunk 1]
```diff
@@ -406,11 +406,12 @@ def read_csv(
                 kdf[col] = kdf[col].astype(dtype)
 
     if squeeze and len(kdf.columns) == 1:
-        return first_series(kdf)
-    return kdf
+        return cast(DataFrame, first_series(kdf))
```
Comment (ueshin) [inline databricks/koalas/namespace.py:409]: This shouldn't be casted. `read_csv` CAN return `Series` when `squeeze is True`.

File: databricks/koalas/namespace.py [hunk 2]
```diff
@@ -1446,7 +1449,7 @@ def read_sql(sql, con, index_col=None, columns=None, **options):
 
 def to_datetime(
     arg, errors="raise", format=None, unit=None, infer_datetime_format=False, origin="unix"
-):
+) -> Union[DataFrame, Series, pd.Timestamp, pd.DatetimeIndex]:
```
Comment (ueshin) [inline databricks/koalas/namespace.py:1452]: Could you remove type annotations for functions which are intended to work with both Koalas and pandas for now?

Currently it fails due to the type inference:

```py
>>> kdf = ks.DataFrame({"year": [2015, 2016], "month": [2, 3], "day": [4, 5], 'ymd': ['20150204', '20160305']})
>>> kdf
   year  month  day       ymd
0  2015      2    4  20150204
1  2016      3    5  20160305
>>> kdf.ymd.transform(ks.to_datetime)
Traceback (most recent call last):
...
TypeError: issubclass() arg 1 must be a class
```

cc @HyukjinKwon

File: databricks/koalas/namespace.py [hunk 3]
```diff
@@ -2179,14 +2182,14 @@ def resolve_func(kdf, this_column_labels, that_column_labels):
...
-def melt(frame, id_vars=None, value_vars=None, var_name=None, value_name="value"):
+def melt(frame, id_vars=None, value_vars=None, var_name=None, value_name="value") -> DataFrame:
     return DataFrame.melt(frame, id_vars, value_vars, var_name, value_name)
 
 
 melt.__doc__ = DataFrame.melt.__doc__
 
 
-def isna(obj):
+def isna(obj) -> Union[DataFrame, Series, bool, np.ndarray]:
```
Comment (ueshin) [inline databricks/koalas/namespace.py:2192]: ditto.

File: databricks/koalas/namespace.py [hunk 4]
```diff
@@ -2268,7 +2271,7 @@ def isna(obj):
 isnull = isna
 
 
-def notna(obj):
+def notna(obj) -> Union[DataFrame, Series, bool, np.ndarray]:
```
Comment (ueshin) [inline databricks/koalas/namespace.py:2274]: ditto.

File: databricks/koalas/namespace.py [hunk 5]
```diff
@@ -2475,7 +2478,7 @@ def merge(
     )
 
 
-def to_numeric(arg):
+def to_numeric(arg) -> Union[DataFrame, Series, int, float, np.ndarray]:
```
Comment (ueshin) [inline databricks/koalas/namespace.py:2481]: ditto.

File: databricks/koalas/namespace.py (hunk 2 above)
Comment (xinrong-databricks) [reply databricks/koalas/namespace.py]: Good catch! Removed.

File: databricks/koalas/namespace.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/namespace.py]: Fixed in 72e5819. Thanks!

**Review by ueshin — APPROVED** [review]
LGTM.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**